backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/KeyValueHandler.java

```diff
@@ -171,4 +172,40 @@ public KeyValueHandler(Executor executor, FutureJdbi jdbi, String fieldType, Str
           query.execute();
         });
   }
+
+  public InternalFuture<Void> updateKeyValue(String entityId, KeyValue kv) {
+    var currentFuture =
+        InternalFuture.runAsync(
+            () -> {
+              if (kv.getKey().isEmpty()) {
+                throw new InvalidArgumentException("Empty key");
+              }
+            },
+            executor);
+
+    currentFuture =
+        currentFuture.thenCompose(
+            unused ->
+                // Update into KV table
+                jdbi.useHandle(
+                    handle -> {
+                      handle
+                          .createUpdate(
+                              "Update "
+                                  + getTableName()
+                                  + " SET kv_key=:key, kv_value=:value, value_type=:type "
+                                  + " where entity_name=:entity_name and field_type=:field_type and kv_key=:key and "
```

Comment:
What is supposed to happen if the given key doesn't exist?

Reply:
Not sure. but what is expected here?? should we throw error if not exists Or we will just add it in database??

Reply:
On old flow if key was not exists then we are logged it as a new entry.

Reply:
No strong preference. We might end up removing this update since ERs don't have them, so they are probably not strictly necessary. That will come in later. Add a TODO for us to revisit

Reply:
Added TODO on the method